File: packages/core/e2e/fixtures/typescript-project/src/parser.ts
/**
 * Code parser for analyzing TypeScript files
 */

export interface ParseResult {
    functions: string[];
    classes: string[];
    imports: string[];
}

export function parseCode(source: string): ParseResult {
    const functions: string[] = [];
    const classes: string[] = [];
    const imports: string[] = [];

    // Simple regex-based parsing for demo
    const functionMatches = source.match(/function\s+(\w+)/g);
    if (functionMatches) {
        functionMatches.forEach(match => {
            const name = match.replace('function ', '');
            functions.push(name);
        });
    }

    const classMatches = source.match(/class\s+(\w+)/g);
    if (classMatches) {
        classMatches.forEach(match => {
            const name = match.replace('class ', '');
            classes.push(name);
        });
    }

    const importMatches = source.match(/import\s+.*?\s+from\s+['"](.+?)['"]/g);
    if (importMatches) {
        importMatches.forEach(match => {
            const path = match.match(/from\s+['"](.+?)['"]/)?.[1];
            if (path) imports.push(path);
        });
    }

    return { functions, classes, imports };
}

export function extractComments(source: string): string[] {
    const comments: string[] = [];
    const lines = source.split('\n');

    for (const line of lines) {
        const commentIndex = line.indexOf('//');
        if (commentIndex !== -1) {
            comments.push(line.slice(commentIndex + 2).trim());
        }
    }

    return comments;
}

File: packages/core/e2e/fixtures/python-project/src/parser.py
"""
Code parser for analyzing Python files
"""

import ast
import re
from typing import List, Dict, Any


def parse_code(source: str) -> Dict[str, List[str]]:
    """Parse Python source code and extract functions, classes, and imports."""
    result = {
        'functions': [],
        'classes': [],
        'imports': []
    }

    try:
        tree = ast.parse(source)
        for node in ast.walk(tree):
            if isinstance(node, ast.FunctionDef):
                result['functions'].append(node.name)
            elif isinstance(node, ast.ClassDef):
                result['classes'].append(node.name)
            elif isinstance(node, ast.Import):
                for alias in node.names:
                    result['imports'].append(alias.name)
            elif isinstance(node, ast.ImportFrom):
                module = node.module or ''
                result['imports'].append(module)
    except SyntaxError:
        pass

    return result


def extract_docstrings(source: str) -> List[str]:
    """Extract docstrings from Python source code."""
    docstrings = []
    try:
        tree = ast.parse(source)
        for node in ast.walk(tree):
            if isinstance(node, (ast.FunctionDef, ast.ClassDef)):
                if ast.get_docstring(node):
                    docstrings.append(ast.get_docstring(node))
    except SyntaxError:
        pass
    return docstrings


def extract_comments(source: str) -> List[str]:
    """Extract comments from Python source code."""
    comments = []
    lines = source.split('\n')
    for line in lines:
        if '#' in line:
            comment = line[line.index('#')+1:].strip()
            if comment:
                comments.append(comment)
    return comments
